pkg/storage/chunk/cache/memcached.go

```diff
@@ -171,11 +178,13 @@ func (c *Memcached) fetch(ctx context.Context, keys []string) (found []string, b
 func (c *Memcached) fetchKeysBatched(ctx context.Context, keys []string) (found []string, bufs [][]byte, missed []string, err error) {
 	resultsCh := make(chan *result)
 	batchSize := c.cfg.BatchSize
-
 	go func() {
 		for i, j := 0, 0; i < len(keys); i += batchSize {
 			batchKeys := keys[i:math.Min(i+batchSize, len(keys))]
 			select {
+			case <-ctx.Done():
+				c.closeAndStop()
+				return
```

Comment:
Good point about fetcher 👍 Actually thought about sending error back didn't end up doing because need some rewrite to make all those go routines return error and make other main goroutine to listen on some `errCh` correctly.

But seeing how fetcher uses it, it's definitely worth sending error up. Agree. will fix it.

Reply:
Done 👍

Reply:
Looks good, thanks 👍